### DISCUSSION TITLE
USB Serial uploading suddenly not working on ATtiny841 (Optiboot)

### DISCUSSION THREAD

**Discussion by CaptainCheeze29** [SpenceKonde/ATTinyCore#867] [Support / Q & A]
So in a previous discussion I laid out the ATtiny841 based Dev board that I've been working on (I'll put schematics and images below). The design worked perfectly (other than some weird pin mapping peculiarities), and code could be easily uploaded using the built in CH340 serial chip after the Optiboot bootloader had been installed by ISP using an Arduino Uno as programmer. 

Upload settings:
![image](https://github.com/SpenceKonde/ATTinyCore/assets/94358321/19c0d821-ab81-447b-a7b7-b77d0586e9ad)

However, just in the last week I've had the same issue happen to two of my 841 chips. I go to upload the code and it gives me the following problem every time:

![ATtiny841_UploadError](https://github.com/SpenceKonde/ATTinyCore/assets/94358321/abeeeab4-bbd6-4d4a-a897-aa9858ff7346)

until this is the final output:

```
Sketch uses 396 bytes (5%) of program storage space. Maximum is 7552 bytes.
Global variables use 9 bytes (1%) of dynamic memory, leaving 503 bytes for local variables. Maximum is 512 bytes.
"C:\Users\sleig\AppData\Local\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino18/bin/avrdude" "-CC:\Users\sleig\AppData\Local\Arduino15\packages\ATTinyCore\hardware\avr\1.5.2/avrdude.conf" -v -pattiny841 -carduino -PCOM4 -b57600 -D "-Uflash:w:C:\Users\sleig\AppData\Local\Temp\arduino\sketches\CBC834A4FB3A225BE10EA82694EB35EF/sketch_jun23a.ino.hex:i"

avrdude: Version 6.3-20201216
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "C:\Users\sleig\AppData\Local\Arduino15\packages\ATTinyCore\hardware\avr\1.5.2/avrdude.conf"

         Using Port                    : COM4
         Using Programmer              : arduino
         Overriding Baud Rate          : 57600
         Setting bit clk period        : 5.0
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 2 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 3 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 4 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 5 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 6 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 7 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 8 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 9 of 10: not in sync: resp=0x4b
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 10 of 10: not in sync: resp=0x4b

avrdude done.  Thank you.

Failed uploading: uploading error: exit status 1
```

I have currently tried the following steps to correct this and nothing has worked.

Reconnecting/changing USB cables
Reinstalling the bootloader via ISP
Reinstalling Arduino IDE
Older versions of IDE
Reinstalling AttinyCore library

As I said, this had been working perfectly up until recently. If anyone has any advice as to what might have changed or where I'm going wrong. Please let me know. Thanks guys.

Dev board schematics and images:
[Attiny841_Schematic.pdf](https://github.com/user-attachments/files/15943880/Attiny841_Schematic.pdf)
![image](https://github.com/SpenceKonde/ATTinyCore/assets/94358321/0a114fb3-4e31-446a-ac7c-4f39e5763d64)
![image](https://github.com/SpenceKonde/ATTinyCore/assets/94358321/eda41470-0eb6-4538-8bdd-ed0e7d436e8c)

**Comment by hmeijdam** [comment]
You have selected USBASP as programmer, but you talk about using an Arduino as ISP.

**Reply by CaptainCheeze29** [reply]
> You have selected USBASP as programmer, but you talk about using an Arduino as ISP.

I only use the ISP when I flash the booloader to the chip initially.

**Comment by hmeijdam** [comment]
Your upload command looks identical to mine. The only thing I can think of is to check your CH340 driver. FWIW, this is the one I have.

![image](https://github.com/SpenceKonde/ATTinyCore/assets/9595465/ec9cda31-d3f8-4352-8504-cf72208ed95e)

**Reply by CaptainCheeze29** [reply]
Thanks. Comparing to my drivers it seems that yours is a newer version. It never stopped me before, but I'll look for an update.

**Comment by hmeijdam** [comment]
I downloaded it from [here](https://www.wch-ic.com/downloads/category/30.html)

**Reply by CaptainCheeze29** [reply]
Unfortunately same issue. But I've just suddenly seen this pop in my IDE for the first time:

![image](https://github.com/SpenceKonde/ATTinyCore/assets/94358321/e6edf37a-0a4d-4b09-b6e5-2f4b387c9c83)

**Reply by hmeijdam** [reply]
You can try [this workaround](https://github.com/SpenceKonde/megaTinyCore/issues/1059#issuecomment-1950286469) for that

**Reply by CaptainCheeze29** [reply]
I've been trying to follow the instructions in that thread, but I'm not sure if I'm doing it right and not having success. Feel a bit like a fool.

**Reply by hmeijdam** [reply]
In your options in your IDE you can set the additional board managers URL's
Replace the drazzy URL by the descartes URL. Then your ide will check for that URL, which has a valid certificate

**Reply by CaptainCheeze29** [reply]
Ok ok ok! I'm getting somewhere now. I've just got the ATtinyCore back up and running. Thank you!

I've now re-burnt the bootloader onto my Rev01 board (which has a usbC connector and CH340G chip) and it's worked. Bootloader went on clean, and a WS2812B rainbow code is up and running perfectly!

Then tried the same thing with my Rev02 board (Usb-mini connector and CH340C chip) and the bootloader goes on no problem, but code is still not uploading...Giving the old original 10 tries and failure issue again.
Could it be something to do with the different CH340 chips? One with an ext crystal and one without? Or maybe with the different usb types?

**Reply by hmeijdam** [reply]
Are you using a USB hub? Try with a direct USB port from your PC.

**Reply by CaptainCheeze29** [reply]
No USB hub. Directly to port on PC. Tried the other ports as well.
